packages/react-components/src/components/Attachment/AttachmentCard.tsx

```diff
@@ -66,27 +67,30 @@ export const _AttachmentCard = (props: _AttachmentCardProps): JSX.Element => {
     return 'progress' in attachment ? attachment.progress : undefined;
   }, [attachment]);
 
-  const isUploadComplete = useMemo(() => {
-    return progress !== undefined && progress > 0 && progress < 1;
+  const isUploadInProgress = useMemo(() => {
+    return progress !== undefined && progress >= 0 && progress < 1;
   }, [progress]);
 
   const [announcerString, setAnnouncerString] = useState<string | undefined>(undefined);
   const localeStrings = useLocaleAttachmentCardStringsTrampoline();
   const uploadStartedString = props.strings?.uploading ?? localeStrings.uploading;
   const uploadCompletedString = props.strings?.uploadCompleted ?? localeStrings.uploadCompleted;
 
-  const showProgressIndicator = progress !== undefined && progress > 0 && progress < 1;
-
   useEffect(() => {
-    if (showProgressIndicator) {
+    if (isUploadInProgress) {
       setAnnouncerString(`${uploadStartedString} ${attachment.name}`);
     } else if (progress === 1) {
       setAnnouncerString(`${attachment.name} ${uploadCompletedString}`);
     } else {
       setAnnouncerString(undefined);
     }
-  }, [progress, showProgressIndicator, attachment.name, uploadStartedString, uploadCompletedString]);
+  }, [progress, isUploadInProgress, attachment.name, uploadStartedString, uploadCompletedString]);
 
+  const extension = useMemo((): string => {
```

Comment:
Here's a breakdown of the pattern:

`(?: ... )`: A non-capturing group that allows us to group parts of the regex without saving the matched substring.
`\.`: Escapes the dot character so that it matches an actual dot, not any character (which is the default meaning of . in regex).
`([^.]+)`: A capturing group that matches one or more characters that are not a dot.
?: Makes the preceding group optional, meaning the regex can match strings with or without a file extension.
`$`: Asserts the position at the end of the line, ensuring that the match is right before the end of the string.


This regex is for extracting the file extension from a path or URL. For example, it would match `jpg` in `image.jpg`.